experimental/client-csharp/src/Wandb/Config.cs

```diff
@@ -0,0 +1,67 @@
+using System.Text.Json;
+using WandbInternal;
+
+namespace Wandb
+{
+    /// <summary>
+    /// Represents a dynamic configuration that allows for setting and retrieving key-value pairs.
+    /// </summary>
+    public class Config
+    {
+        /// <summary>
+        /// Occurs when a configuration item is updated.
+        /// </summary>
+        public event Action<string, object> ConfigUpdated;
```

Comment:
It's best to avoid using events when you don't have to. I don't think `ConfigUpdated` should be part of the public API. `Config` should hold a reference to a `Run` and interact with it directly. This is important for correctly implementing the config getter for resumed runs.

Using an event also forces the callback to be `async void`, which is problematic (https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void). Socket exceptions during `run.Config["key"] = value` will be swallowed instead of bubbling up on that line; we're also pretending that this method runs synchronously when it's really async: after `run.Config["key"] = value`, the config might not have been updated!

Pretending the config is a dictionary worked in Python because that API is synchronous. The C# API needs to be explicit about this. We might as well drop the `Config` type and replace it with `Run` methods:

```
await run.UpdateConfig(myConfigDictionary)
```

Reply:
will something like `run.Config.Update(...)` would be better?

Reply:
theoretically we also need to support `Remove`?

Reply:
Using a separate `Config` object would allow users to pass it between functions, like

```cs
async Task _setRunConfig(Wandb.Config config) {
    await config.Update(...)
}

await _setRunConfig(run.Config);
```

but that can be accomplished by passing the run itself:

```cs
async Task _setRunConfig(Wandb.Run run) {
    await run.UpdateConfig(...)
}

await _setRunConfig(run);
```

We shouldn't use a `Config` property just for namespacing `Run` methods. I don't think it helps from a "modeling" perspective either: we don't have any other methods that return a `Config` or any methods at all that accept one; it's not useful to think of a `Config` separately from a `Run`.